airsonic-main/src/main/java/org/airsonic/player/service/search/SearchServiceUtilities.java

```diff
@@ -71,6 +79,47 @@
     @Autowired
     private MediaFileService mediaFileService;
 
+    private static final Logger LOG = LoggerFactory.getLogger(SearchServiceUtilities.class);
+
+    private Random random;
+
+    {
+
+        String algorithmNative = "NativePRNG";
+        String algorithmSHA = "SHA1PRNG";
+
+        List<String> algorithms = Arrays.asList(Security.getProviders()).stream()
+                .flatMap(p -> p.getServices().stream().filter(s -> "SecureRandom".equals(s.getType())))
+                .map(s -> s.getAlgorithm()).collect(Collectors.toList());
+
+        if (algorithms.contains(algorithmNative)) {
+            /* linux */
+            try {
+                random = SecureRandom.getInstance(algorithmNative);
+                LOG.info("NativePRNG is used to create a random list of songs.", algorithmNative);
```

Comment:
I would probably make all these `LOG.info` messages be `LOG.debug`, as most users won't care about having log messages about what PRNG is in use.